src/components/ThreejsEditorModal.jsx

```diff
@@ -134,6 +161,54 @@ export class ThreejsEditorModal extends ModalDialog {
             },
             false,
         );
+
+        document.addEventListener("mousedown", (event) => {
+            // on right click and hold disable orbit controls to allow pan
+            if (event.button === THREE.MOUSE.RIGHT) {
+                this.editor.controls.enabled = false;
+                return;
+            }
+
+            // on a left click detect if the click was on the object and disable orbit controls in that case
+            if (event.button === THREE.MOUSE.LEFT) {
+                // calculate mouse position in normalized device coordinates
+                // https://threejs.org/docs/#api/en/core/Raycaster
+
+                this.mouse.x = (event.clientX / window.innerWidth) * 2 - 1;
+                this.mouse.y = -(event.clientY / window.innerHeight) * 2 + 1;
+
+                this.raycaster.setFromCamera(this.mouse, this.editor.camera);
+
+                const objects = this.editor.scene.children;
+                const intersects = this.raycaster.intersectObjects(objects, true);
+                this.editor.controls.enabled =
+                    intersects.length === 0 && !isMultipleSelectionActive();
+            }
+        });
+
+        // Disable rotation and set orbit controls target to the selected object if it's a Mesh (Atom)
+        this.editor.signals.objectSelected.add((selectedObject) => {
+            if (selectedObject && selectedObject.type === "Mesh") {
+                this.editor.controls.target.copy(selectedObject.position);
+                this.editor.controls.enabled = false;
+            } else {
+                this.editor.controls.enabled = true;
+            }
+        });
+        // on signal of object changing position disable orbit controls
+        this.editor.signals.objectChanged.add((selectedObject) => {
+            if (selectedObject && selectedObject.type === "Mesh") {
+                this.editor.controls.enabled = false;
+            }
+        });
+
+        // on right click release enable orbit controls after pan
+        document.addEventListener("mouseup", (event) => {
+            if (event.button === THREE.MOUSE.RIGHT) {
+                this.editor.controls.enabled = true;
+            }
+        });
+
```

Comment:
Since all of these events are listening to controls we can move them to the mixin, probably